### PR TITLE
TT-5376 HTTPServerOptions Certificate custom env decoder

### PR DESCRIPTION
**Pull request by tbuchaillot**
Fixes https://tyktech.atlassian.net/browse/TT-3795

## Description

Added a custom decoder for TYK_GW_HTTPSERVEROPTIONS_CERTIFICATES since it is not supported by https://github.com/kelseyhightower/envconfig. In the custom decoder we just simply json unmarshal the struct.

Example:
```
export TYK_GW_HTTPSERVEROPTIONS_CERTIFICATES=[{\"domain_name\":\"test\",\"cert_file\":\"/etc/ssl/certs/tyk.local.crt\",\"key_file\":\"keyfiletest\"}]
 ```

## Related Issue

https://stackoverflow.com/c/tyk-technologies/questions/2134
https://tyktech.atlassian.net/browse/TT-5376
## Motivation and Context

Make all the gateway configurable by env vars.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

**Comment by letzya** [comment]
@buger any idea why `certificates ` is deprecated at all? maybe we can remove the deprecation?

File: config/config.go [hunk 1]
```diff
@@ -326,6 +326,15 @@ type LocalSessionCacheConf struct {
 	CachedSessionTimeout int `json:"cached_session_timeout"`
 	CacheSessionEviction int `json:"cached_session_eviction"`
 }
+type CertsData []CertData
+
+func (certs *CertsData) Decode(value string) error {
+	err := json.Unmarshal([]byte(value), certs)
+	if err != nil {
+		log.Info("Error unmarshaling TYK_GW_HTTPSERVEROPTIONS_CERTIFICATES")
```
Comment (furkansenharputlu) [inline config/config.go:334]: log.Error

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3716/merge
Commit: 43129fc250d73d62e359bc48153d4ab517685a95 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2004353660)

File: config/config.go (hunk 1 above)
Comment (tbuchaillot) [reply config/config.go]: Done!

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3716/merge
Commit: df6cbeefe6ce5201350ddbd142247614a040b398 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2345976719)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3716/merge
Commit: 0472d25d310262a548e15b4572d310ffd504a9fa 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2345988965)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3716/merge
Commit: 5aa086f64b6ac6a51d4dd85590920acc7aef09c1 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2345983229)

**Review by matiasinsaurralde — APPROVED** [review]
Looks good

**Comment by tbuchaillot** [comment]
> @buger any idea why `certificates ` is deprecated at all? maybe we can remove the deprecation so it'll match the dashboard. @zalbiraw will give more details.

@letzya we are also adding the `ssl_certificates` configuration option on the dashboard side. Despite that, we do think that it would be good to have both options available with env vars, as we have in JSON configs.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3716/merge
Commit: 22284996e4fc885490cbba6224a60c3418de57ca 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2417454922)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3716/merge
Commit: 1c43e710033bbc2e78029f50d6da5087124e396b 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2420216273)

**Review by letzya — APPROVED** [review]

**Pull request merged**